### PR TITLE
[README] Update README for third-part library installation

### PR DESCRIPTION
**Pull request by chhzh123**
## Description ##
This PR reorganizes the README structure for benchmarking, which provides necessary instructions for users to run the test scripts in a brand-new environment. To reduce duplication, only the README file under the benchmark folder describes the installation requirements. Other README just points to this file.

It also fixes a tiny issue of Conv1DWithSyncFunc that may cause import errors when the environment does not have the transformer library.

cc @comaniac

### CODE REVIEW

File: benchmark/README.md [hunk 1]
```diff
@@ -3,10 +3,111 @@
...
 
-## Single Node with Tensor Parallelism on Megatron
+## Preparation
+
+### Install Dependencies
+
+Please use the following command to install the dependencies. By default, [PyTorch](https://pytorch.org/) 2.0 is installed for better performance.
+
+```bash
+pip3 install torch torchvision transformers datasets matplotlib tabulate networkx triton pybind11
```
Comment (comaniac) [inline benchmark/README.md:15]: nit: This might install the wrong pytorch in terms of CUDA version.

File: slapo/sharding/shard_ops.py [hunk 2]
```diff
@@ -548,7 +548,7 @@ class ShardConv1D(ShardMethod):
     and returns the output type (partial or partition) after sharding.
     """
 
-    class Conv1DWithSyncFunc(Conv1D):
+    class Conv1DWithSyncFunc(Conv1D if Conv1D is not None else nn.Module):
```
Comment (comaniac) [inline slapo/sharding/shard_ops.py:551]: If transformers is not available we should not run this op anyways, so I would suggest raising an exception in its constructor.

**Review by comaniac — APPROVED** [review]
LGTM

File: benchmark/README.md (hunk 1 above)
Comment (chhzh123) [reply benchmark/README.md]: Added clarification

File: slapo/sharding/shard_ops.py (hunk 2 above)
Comment (chhzh123) [reply slapo/sharding/shard_ops.py:551]: Fixed

**Pull request merged**